plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -5682,6 +5684,36 @@ public void testNestedColumnWithDuplicateName()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testExternalLocationWithTrailingSpace()
+            throws IOException
+    {
+        String tableName = "test_external_location_with_trailing_space_" + randomNameSuffix();
+        assertUpdate(format("CREATE TABLE %s (a varchar, b varchar) WITH (format='PARQUET')", tableName));
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('a', 'b')", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String tableLocationWithTrailingSpace = tableLocation + " ";
+
+        // Move data to directory with trailing space
+        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(getSession().toConnectorSession());
+        fs.renameFile(tableLocation, tableLocationWithTrailingSpace);
+
+        String newTableName = "test_external_location_with_trailing_space_" + randomNameSuffix();
```

Comment:
You could create a table with `external_location` from the beginning instead of going through a managed table. See `io.trino.plugin.hive.BaseHiveConnectorTest#testCreateExternalTable(java.lang.String, java.lang.String, java.lang.String, java.util.List<java.lang.String>)`